src/CommonITILObject.php

```diff
@@ -1115,6 +1115,16 @@ public function prepareInputForUpdate($input)
             }
         }
 
+        $can_assign_slm = Session::haveRight(SLM::$rightname, SLM::RIGHT_ASSIGN);
```

Comment:
Unless I am wrong, this will be done after rules processing, so it will not be possible to affect SLA/OLA using rules if ticket is created by a user that has no right to define it.